### PR TITLE
OID4VCI Integration

### PR DESCRIPTION
**Pull request by theosirian**
Depends on https://github.com/spruceid/oid4vci-rs/pull/18.

Bindings compilation status:
- [x] iOS
- [x] Android

### CODE REVIEW

File: Cargo.toml
Comment (sbihel) [inline Cargo.toml:24]: You can import the whole of `ssi` in this project

File: src/oid4vci/http_client.rs
Comment (sbihel) [inline src/oid4vci/http_client.rs:124]: IMO let's not worry about doing HTTP calls from the native code, and just use `oid4vci::openidconnect::reqwest::Client`

**Review by sbihel — COMMENTED** [review]
The CI is failing because of async (which I've fixed in https://github.com/spruceid/mobile-sdk-rs/pull/25)

There's more code than I was expecting but it all looks good. Maybe in the future we'll realise it doesn't need to be too flexible.

I won't approve until the CI is passing and it's all tested end-to-end

File: src/oid4vci/http_client.rs
Comment (theosirian) [reply src/oid4vci/http_client.rs:124]: I went with a trait because it'll be useful to let the library user potentially configure different parameters for an HTTP client that are not specific to the OID4VCI spec, like extra headers, timeouts, certificates, etc.
But we can definitely have `oid4vci::openidconnect::reqwest::Client` be the default.

Comment (sbihel) [reply src/oid4vci/http_client.rs:124]: Agreed. We could offer a default with `reqwest` but let's do that in a separate PR if we find the need for it. It doesn't seem hard to implement the HTTP client in Swift/Kotlin and offers better debugging capabilities.

File: src/oid4vci/proof_of_possession.rs
Comment (Ryanmtate) [inline src/oid4vci/proof_of_possession.rs:54]: #28 has an implementation of a `[KeyManager](https://github.com/spruceid/mobile-sdk-rs/pull/28/files#diff-741fdf60909022c1baae65c0125d16f0a2ed8b4f2bcab2a25eb5ec1ba265912b)` callback interface that may be worth exploring for use here.

File: src/oid4vci/mod.rs
Comment (cobward) [inline src/oid4vci/mod.rs:47]: What's the format of this string? The code implies that it's JSON, but a credential offer is parsed from a URI. Could we just pass the URI here instead?

**Review by cobward — COMMENTED** [review]
I would also like to see some consideration for the VdcCollection and key management. How does a credential get stored in the VdcCollection, and how is that associated with a keypair that can be accessed by credential presentation code.

**Comment by Ryanmtate** [comment]
@sbihel @cobward 

After syncing with @theosirian, we're going to try to merge this as-is before rebasing on the open draft PRs.

We'll upstream the changes for the `KeyManager` and the `common` types after this is merged.

Would it be okay to get this final review today for merge?

**Review by sbihel — APPROVED** [review]

**Pull request merged**
